### PR TITLE
Adds ability to configure S3 object stores via environment variables

### PR DESCRIPTION
**Pull request by pierreozoux**
closes #1193 #1124 #1134

Co-authored-by: Adam Jenkins <adamjenkins1701@gmail.com>
Co-authored-by: Stephen Cuppett <scuppett@redhat.com>

Ok, back to square 0.. The build test did fail with this one.

https://github.com/nextcloud/docker/actions/runs/228749540

I tried locally, and I have a server error on the fpm, it looks like this:

```
{"reqId":"zXfPoqSHkI3gHebpzbIy","level":3,"time":"2020-08-28T14:49:55+00:00","remoteAddr":"","user":"--","app":"base","method":"GET","url":"--","message":{"Exception":"ErrorException","Message":"session_name(): Cannot change session name when headers already sent","Code":0,"Trace":[{"function":"trapError","class":"OC\\Session\\Internal","type":"->","args":[2,"session_name(): Cannot change session name when headers already sent","/var/www/html/lib/private/Session/Internal.php",211,{"functionName":"session_name","parameters":["oci0pb5zcgsr"],"silence":false}]},{"function":"session_name","args":["oci0pb5zcgsr"]},{"file":"/var/www/html/lib/private/Session/Internal.php","line":211,"function":"call_user_func_array","args":["session_name",["oci0pb5zcgsr"]]},{"file":"/var/www/html/lib/private/Session/Internal.php","line":57,"function":"invoke","class":"OC\\Session\\Internal","type":"->","args":["session_name",["oci0pb5zcgsr"]]},{"file":"/var/www/html/lib/base.php","line":430,"function":"__construct","class":"OC\\Session\\Internal","type":"->","args":["oci0pb5zcgsr"]},{"file":"/var/www/html/lib/base.php","line":648,"function":"initSession","class":"OC","type":"::","args":[]},{"file":"/var/www/html/lib/base.php","line":1094,"function":"init","class":"OC","type":"::","args":[]},{"file":"/var/www/html/index.php","line":36,"args":["/var/www/html/lib/base.php"],"function":"require_once"}],"File":"/var/www/html/lib/private/Session/Internal.php","Line":187,"CustomMessage":"--"},"userAgent":"--","version":""}
```

@cuppett and @adamjenkins1 if you know how to fix? I'd be glad :)

thanks again for your help!

### CODE REVIEW

**Comment by cuppett** [comment]
There's a blank line at the top of the s3.config.php before the <?. If you remove that you'll be fine.

File: .config/s3.config.php [hunk 1]
```diff
@@ -0,0 +1,22 @@
+
```
Comment (cuppett) [inline .config/s3.config.php:1]: Oof. Lose this! :)

Comment (pierreozoux) [inline .config/s3.config.php:1]: > 
```suggestion
```

Comment (pierreozoux) [reply .config/s3.config.php]: Perfect! Thanks :+1: 

I didnt know one line was so powerful ;)

**Pull request merged**